### PR TITLE
Handle `family` passed as keyword argument

### PR DESCRIPTION
**Pull request by emontnemery**
Handle `family` passed as keyword argument to `socket` constructor

### CODE REVIEW

**Comment by emontnemery** [comment]
@miketheman can this get merged, or are there any concerns?

Instead of this:
```
def __new__(cls, *args, **kwargs):
```

We could perhaps do:
```py
def __new__(cls, family=AF_INET, type=SOCK_STREAM, proto=0, fileno=None)
```

**Comment by miketheman** [comment]
Hi @emontnemery !

I'm lacking context as to why this is an improvement on the existing behavior, as well as any tests to validate the change works as intended.

File: pytest_socket.py [hunk 1]
```diff
@@ -86,7 +86,10 @@ class GuardedSocket(socket.socket):
         """ socket guard to disable socket creation (from pytest-socket) """
         def __new__(cls, *args, **kwargs):
```
[the remark this answers is not in the document]
Comment (miketheman) [reply pytest_socket.py:87]: Interesting - maybe this should be more compatible with the defaults of the underlying class? https://github.com/python/cpython/blob/59435eea08d30796174552c0ca03c59b41adf8a5/Lib/socket.py#L220

[the remark this answers is not in the document]
Comment (emontnemery) [reply pytest_socket.py:87]: OK, let's mimic the CPython implementation, i.e.:
```py
def __new__(cls, family=-1, type=-1, proto=-1, fileno=None)
```

**Comment by miketheman** [comment]
Thanks for working through this!

**Pull request merged**